tests/integration/utils.go

```diff
@@ -2565,3 +2571,26 @@ func parseCreateDocs(action CreateDoc, collection client.Collection) ([]*client.
 		return []*client.Document{val}, nil
 	}
 }
+
+func checkNodesIdentities(s *state) {
+	for i, node := range s.nodes {
+		var actualIdent immutable.Option[identity.PublicRawIdentity]
+		err := withRetryOnNode(
+			node,
+			func() error {
+				var err error
+				actualIdent, err = node.GetNodeIdentity(s.ctx)
+				return err
+			},
+		)
+		require.NoError(s.t, err, s.testCase.Description)
+
+		expectedIdent := getIdentity(s, i, immutable.Some(0))
+		expectedRawIdent := immutable.None[identity.PublicRawIdentity]()
+		if expectedIdent.HasValue() {
+			expectedRawIdent = immutable.Some(expectedIdent.Value().IntoRawIdentity().Public())
+		}
+
+		require.Equal(s.t, expectedRawIdent, actualIdent, "identities mismatch")
+	}
+}
```

Comment:
question: I don't quite understand why this is needed / what this actually does. I also feel like we are "mixing" identities? Isn't a node's `Node identity` different from a nodes `Requesting Identities` (which we have set up from before). I don't get the correlation.

Reply:
Ah cheers for explaining!

Here is my suggestion regarding resolving the `mixing` of identities as you have right now:
- Rename `getIdentity` to `getRequestingIdentity` and then make a similar function called `getNodeIdentity`.
- Similarly rename the `identities [][]identity.Identity` on `state` struct (i.e. `s.idenities`) to `requestingIdentities` and add a new variable to the `state` struct called `nodeIdentities [][]identity.Identity` which does similar stuff for every node's identity.

Then you can do what you are doing above and it will be somewhat easier to understand, would still document this in the `checkNodesIndentity` function.

question: Could all this checking not just happen in the assertion function that tests the `GetNodeIdentity` behavior?

Reply:
I already have the identities separated, but it's on another branch. I call them `nodeIdentities` and `userIdentities`. You think I should cherry-pick it here?

> question: Could all this checking not just happen in the assertion function that tests the GetNodeIdentity behavior?

you mean something like this?
```go
testUtils.AssertNodeIdentity {
    NodeID: 0,
    ExpectedIdentity: 0,
}
```
I'm not sure it's worth the effort. And I don't see how it will improve the test.

Reply:
> I already have the identities separated, but it's on another branch. I call them `nodeIdentities` and `userIdentities`. You think I should cherry-pick it here?

Interesting, why is it in a separate PR? I would really like us to not mix both these different kinds of identities. I do slightly prefer the name `requestingIdentities` or `requestorIdentities` over `userIdentities` because incase of relationship management we take an other argument that is the `target actor's identity` which can be confused as a `userIdentity`.

> ```go
> testUtils.AssertNodeIdentity {
>     NodeID: 0,
>     ExpectedIdentity: 0,
> }
> ```
> 
> I'm not sure it's worth the effort. And I don't see how it will improve the test.


As for the test it's an interesting one for sure I am not sure off the top of my head what the best approach might be, but here are 4 test cases that I would like to see definitely see tested:

- If no `node-identity-key` is not given on start then
	- keys are randomly generated (can just assert it is non-empty)
	- the keys stay the same even after database restart (can assert that the key from the above case is same as after restart)
	
- If `node-identity-key` is given on start then
	- keys are actually set based on that given key
	- the keys stay the same even after database restarts to that same key.
	
If you feel that setting integration tests might be a bit useless and not worth the effort I can also vibe with just integration tests asserting the above 4 test cases (perhaps might be easier). However maybe others might have other thoughts on this. @AndrewSisley ?

Reply:
> Interesting, why is it in a separate PR? I would really like us to not mix both these different kinds of identities. I do slightly prefer the name requestingIdentities or requestorIdentities over userIdentities because incase of relationship management we take an other argument that is the target actor's identity which can be confused as a userIdentity.

It's not 100% ready yet and I wasn't sure it's something we really want. I think it makes sense to create a draft after we merge this one so that you can take a quick look before I go far with it

Reply:
> It's not 100% ready yet and I wasn't sure it's something we really want. I think it makes sense to create a draft after we merge this one so that you can take a quick look before I go far with it

Ah cool. Happy to have it in separate PR. Is definitely best IMO to keep request's identities used for acp separate from the node's identity. Looking forward for that draft PR.

Reply:
Wondering what about the test cases I mentioned, would those be after in that PR?

Reply:
as discussed, created a task to cover `start` command with tests.